cmd/lncli/invoicesrpc_active.go

```diff
@@ -212,12 +215,11 @@ func addHoldInvoice(ctx *cli.Context) error {
 
 	args = args.Tail()
 
-	switch {
-	case ctx.IsSet("amt"):
-		amt = ctx.Int64("amt")
-	case args.Present():
-		amt, err = strconv.ParseInt(args.First(), 10, 64)
+	amt := ctx.Int64("amt")
+	amtMsat := ctx.Int64("amt_msat")
 
+	if amt == 0 && amtMsat == 0 && args.Present() {
```

Comment:
Actually, we should use `IsSet` here, since it will distinguish from attempting to create a zero value invoice

Reply:
Amended my patch to use IsSet.